lib/puma/plugin/telemetry/transforms/l2met_transform.rb

```diff
@@ -0,0 +1,20 @@
+# frozen_string_literal: true
+
+module Puma
+  class Plugin
+    module Telemetry
+      module Transforms
+        # L2Met (Logs to Metrics) transform that makes all keys a `sample#` in the L2Met format.
+        #
+        class L2metTransform
+          def self.call(telemetry)
+            telemetry.transform_keys { |k| "sample##{k}" }.tap do |data|
+              data['name'] = 'Puma::Plugin::Telemetry'
+              # data['source'] = ??
```

Comment:
L2Met allows a special "source" key, and I'm not sure of the best way to handle that. For example, if deployed on Heroku, you could image `data['source'] ||= ENV.fetch('DYNO', nil)` or something like that.

Reply:
I ended up adding in a simple mechanism that will fetch `L2MET_SOURCE` or `DYNO` from `ENV`, falling back to `$PROGRAM_NAME`. Seems a sensible place to start.